### DISCUSSION TITLE
shiny: with quarto (1.3) with rmarkdown (2.14)

### DISCUSSION THREAD

**Discussion by thomasmanke** [quarto-dev/quarto-cli#6086] [Q&A] [answered]
### Description

1. I'm trying to reproduce the "Old Faithful" example with RStudio (2023.03), quarto (1.3)  and rmarkdown (2.14).
(Notice that I installed both quarto and r-rmarkdown using mamba and opened the R project in a dedicated conda environment.) The quarto web page (https://quarto.org/docs/interactive/shiny/) suggests rmarkdown 2.10 is sufficient, but I get the following error:

Error in if (xfun::is_blank(options$code)) { : 
  the condition has length > 1
Calls: .main ... process_group.block -> call_block -> as.strict_list -> hook

2. I suspected an issue with rmarkdown and upgraded (2.14 -> 2.22).  I also followed mamba suggestion and  had to _downgrade_ quarto (1.3 ->1.2) reluctantly. Afterwards the shiny app worked as intended.

My question: Is there any way to develop shiny apps using qmd but with the latest quarto v1.3+.

Thank you very much, Thomas 

----

> sessionInfo()
R version 4.2.3 (2023-03-15)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: macOS Monterey 12.6.3

Matrix products: default
LAPACK: /Users/manke/miniconda3/envs/web/lib/libopenblasp-r0.3.21.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] shiny_1.7.4

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10      digest_0.6.31    later_1.3.0      mime_0.12       
 [5] R6_2.5.1         lifecycle_1.0.3  xtable_1.8-4     magrittr_2.0.3  
 [9] evaluate_0.20    rlang_1.1.0      cli_3.6.1        promises_1.2.0.1
[13] rstudioapi_0.14  ellipsis_0.3.2   rmarkdown_2.14   tools_4.2.3     
[17] httpuv_1.6.9     xfun_0.38        yaml_2.3.7       fastmap_1.1.1   
[21] compiler_4.2.3   htmltools_0.5.5  knitr_1.42

---

Just for completeness, here is the qmd-file (as copied from: https://quarto.org/docs/interactive/shiny)

---
title: "Old Faithful"
format: html
execute:
  echo: false
server: shiny
---

```{r}
sliderInput("bins", "Number of bins:", 
            min = 1, max = 50, value = 30)
plotOutput("distPlot")
```

```{r}
#| context: server
output$distPlot <- renderPlot({
  x <- faithful[, 2]  # Old Faithful Geyser data
  bins <- seq(min(x), max(x), length.out = input$bins + 1)
  hist(x, breaks = bins, col = 'darkgray', border = 'white')
})
```

**Comment by mcanouil** [comment, marked as answer]
I can't reproduce using the latest version of everything.
May I suggest for you to upgrade your R packages?

**Reply by thomasmanke** [reply]
Thank you. Indeed, update.packages() yielded the latest rmarkdown=2.23 and did the trick!
As a confusing aside; to run the shiny app, Rstudio insisted on "upgrading" markdown=3.4.3 (installed by mamba) to 1.7.

It appears my mistake was to rely exclusively on mamba for installation of R packages (an example env.yml is shown below).
Given my other dependencies in this simple example, mamba reported conflicts for anything higher than rmarkdown=2.14 (which resulted in the reported error).

a) I'm concerned that after update.packages() the version numbers reported by 'mamba env list' and those used by R are out of sync. Would you recommend installing R-packages exclusively with R rather than conda/mamba?

b) I still feel that the note on rmarkdown version should be updated, since 2.10 is not sufficient: 
https://quarto.org/docs/interactive/shiny/index.html

---
env.yml
---
name: test
channels:
  - conda-forge
 
dependencies:
  - quarto=1.3
  - r-base=4.2.3
  - r-shiny=1.7.4
  - markdown
  - r-rmarkdown

**Reply by mcanouil** [reply]
In your case, I would suggest to use `renv`.
